src/components/lib/posix/posix.c

```diff
@@ -475,36 +495,36 @@ syscall_emulation_setup(void)
 		cos_syscalls[i] = 0;
 	}
 
-	libc_syscall_override((cos_syscall_t)cos_write, __NR_write);
-	libc_syscall_override((cos_syscall_t)cos_writev, __NR_writev);
-	libc_syscall_override((cos_syscall_t)cos_ioctl, __NR_ioctl);
-	libc_syscall_override((cos_syscall_t)cos_brk, __NR_brk);
-	libc_syscall_override((cos_syscall_t)cos_mmap, __NR_mmap2);
-	libc_syscall_override((cos_syscall_t)cos_munmap, __NR_munmap);
-	libc_syscall_override((cos_syscall_t)cos_madvise, __NR_madvise);
-	libc_syscall_override((cos_syscall_t)cos_mremap, __NR_mremap);
+	libc_syscall_override((cos_syscall_t)(void*)cos_write, __NR_write);
```

Comment:
Feel like these might just be hiding a more serious type mismatch???

Reply:
Problem here is that we don't want to see compiler warnings. This causes tons of warnings. Each syscall function here has a different type, and cannot just use (cos_syscall_t) to cast it (will still have warnings).